### ISSUE TITLE
"invalid type" error should show field name when deserializing with serde

### ISSUE THREAD

**Issue by netvl** [alexcrichton/toml-rs#132]
Currently if a field value is of invalid type, the serde decoder will return an error without the field name: `invalid type: string`. If a TOML document is large and contains multiple values, it becomes hard to find which one is problematic. I think the error should contain the field name.

**Comment by alexcrichton** [comment]
Hm in theory we do this (and try to go to great lengths to do so), so this is surprising! Could you give an example where this shows up?

**Comment by netvl** [comment]
Yep, sure. The following code:
```rust
extern crate serde;
#[macro_use] extern crate serde_derive;
extern crate toml;

use serde::Deserialize;

#[derive(Deserialize, Debug)]
enum ChangeMode {
    #[serde(rename = "sequential")]
    Sequential,
    #[serde(rename = "random")]
    Random,
}

#[derive(Deserialize, Debug)]
struct Root {
    mode: ChangeMode,
}

fn main() {
    const DOCUMENT: &'static str = r#"mode = "random""#;

    let mut parser = toml::Parser::new(DOCUMENT);
    let value = toml::Value::Table(parser.parse().unwrap());

    match Root::deserialize(&mut toml::Decoder::new(value)) {
        Ok(root) => println!("{:#?}", root),
        Err(e) => println!("Error: {}\n{:#?}", e, e),
    }
}
```
The `field` field is apparently empty. Note that this code also illustrates #133.

**Comment by alexcrichton** [comment]
Weird, the `mode` here should definitely be getting extracted. This *may* have to do with rustc-serialize vs serde, but I'll have to dig into this.

**Comment by alexcrichton** [comment]
I've rewritten the parser in the serde 0.9 rewrite https://github.com/alexcrichton/toml-rs/pull/137 and I believe this is handled

**Comment by alexcrichton** [comment]
Fixed in #137
